src/agent/custom-http-agent.ts

```diff
@@ -55,9 +62,14 @@ export class CustomHttpAgent {
     arg,
     canisterId,
     method: methodName,
-    nonce
-  }: Omit<IcrcCallCanisterRequestParams, 'sender'>): Promise<CustomHttpAgentResponse> => {
+    nonce,
+    sender
+  }: IcrcCallCanisterRequestParams): Promise<CustomHttpAgentResponse> => {
+    const hash = await generateHash({canisterId, sender, method: methodName, arg, nonce});    
+    const ingressExpiry = this.getIngressExpiry(hash);
```

Comment:
We probably need a clean-up of the cache before or after the function that removes all entries that are expired to avoid the cache to grow indefinitely.